### DISCUSSION TITLE
HGTconv error

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#5791] [Q&A]
I am not sure why this bug is happening.
I thought it was because batch norm somewhere in HGTconv. Its possible that node_types are empty, that is why i put == False. I don't know if its working or not either.

![image](https://user-images.githubusercontent.com/51344466/197050039-759be5b7-db40-4e93-ae5e-70ebac1e78b8.png)

![image](https://user-images.githubusercontent.com/51344466/197049959-f89244c8-3b17-4e23-9fb2-b6bb94508f25.png)

**Comment by rusty1s** [comment]
How do you initialize `HGT`? Do you have a minimal example to reproduce?

**Reply by DogitoErgoSum** [reply]
i believe it is because of this
![image](https://user-images.githubusercontent.com/51344466/197262619-68c3c8b1-ef14-472c-8d87-8f9584104013.png)
D = self.out_channels /self.heads
If i put i let ; self.out_channels= 71 and self.heads=2 it gives error
but if put self.out_channels= pair , the error disappears

This code was taken from the hgtconv: https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/hgt_conv.html

**Reply by DogitoErgoSum** [reply]
It seems that self.out_channels % self.heads == 0 in order to not give the error.

**Reply by rusty1s** [reply]
Yes, this is correct. I added a basic check to ensure this, see https://github.com/pyg-team/pytorch_geometric/pull/5802.

**Reply by DogitoErgoSum** [reply]
@rusty1s  Is there a way to access the HGTConv attention coefficients?

**Reply by rusty1s** [reply]
Currently no, we would need to add support for this similar to as we do in `GATConv`. Let me know if you are interested in contributing this :)

**Reply by DogitoErgoSum** [reply]
Perhaps in the near future :)
